### PR TITLE
Scalastyle

### PR DESCRIPTION
**Pull request by alexduf**
Enable scalastyle as a test step, and fix the dynamodb crashing the jvm when running testQuick and testOnly.

Let's agree that if it gets in our way we either modify the xml file or get rid of it completely.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -17,7 +19,14 @@ lazy val common = project
       "com.amazonaws" % "aws-java-sdk" % "1.9.31",
       "com.gu" %% "configuration" %  "4.1"
     ),
-    test in Test <<= (test in Test).dependsOn(DynamoDBLocal.Keys.startDynamoDBLocal)
+    test in Test <<= (test in Test).dependsOn(DynamoDBLocal.Keys.startDynamoDBLocal),
+    testOnly in Test <<= (testOnly in Test).dependsOn(DynamoDBLocal.Keys.startDynamoDBLocal),
+    testQuick in Test <<= (testQuick in Test).dependsOn(DynamoDBLocal.Keys.startDynamoDBLocal),
+    scalastyleFailOnError := true,
+    testScalastyle := org.scalastyle.sbt.ScalastylePlugin.scalastyle.in(Compile).toTask("").value,
```
Comment (agulowaty) [inline build.sbt]: 1. shouldn't be `ScalastylePlugin.scalastyle.in(Test)` instead ?
2. how about importing org.scalastyle.sbt.ScalastylePlugin ?

Comment (alexduf) [inline build.sbt]: 1. No, we're checking the style of the production code
2. You're right I'll make it more pretty

**Comment by agulowaty** [comment]
:+1:

**Comment by davidfurey** [comment]
:+1:

**Pull request merged**